### PR TITLE
SWORD25: Add hebrew support

### PR DESCRIPTION
**Pull request by ShaharAriel**
changes:

* use unicode-bidi for hebrew text
* add detection of hebrew patch to detection table

### CODE REVIEW

File: engines/sword25/gfx/graphicengine.cpp [hunk 1]
```diff
@@ -453,4 +456,8 @@ bool GraphicEngine::unpersist(InputPersistenceBlock &reader) {
 	return reader.isGood();
 }
 
+bool GraphicEngine::isRTL(){
```
Comment (sev-) [inline engines/sword25/gfx/graphicengine.cpp:459]: This doesn't follow our code formatting guidelines. Write `bool GraphicEngine::isRTL() {`

File: engines/sword25/detection_tables.h [hunk 2]
```diff
@@ -105,6 +105,15 @@ static const ADGameDescription gameDescriptions[] = {
 		ADGF_NO_FLAGS,
 		GUIO1(GUIO_NOASPECT)
 	},
+	{
+			"sword25",
+			"",
+			AD_ENTRY1s("lang_he.b25c", "7b0e58798793e0dd87023ef112a11d0f", 461034),
```
Comment (sev-) [inline engines/sword25/detection_tables.h:111]: Where can I get this file?

File: engines/sword25/gfx/graphicengine.h [hunk 3]
```diff
@@ -231,8 +231,8 @@ class GraphicEngine : public ResourceService, public Persistable {
 	// -------------------
 	bool persist(OutputPersistenceBlock &writer) override;
 	bool unpersist(InputPersistenceBlock &reader) override;
-
-	static void ARGBColorToLuaColor(lua_State *L, uint color);
+    bool isRTL();
+    static void ARGBColorToLuaColor(lua_State *L, uint color);
```
Comment (sev-) [inline engines/sword25/gfx/graphicengine.h:235]: For whatever reason, you changed the indentation here from tabs to spaces. Please revert.

Also, please fix the previous line too.

**Review by sev- — COMMENTED** [review]
Good stuff, but I have a few minor notes.

File: engines/sword25/detection_tables.h (hunk 2 above)
Comment (ShaharAriel) [reply engines/sword25/detection_tables.h:111]: The file will be release after this PR merge in public facebook group (of adventure game fans) [in this group ](https://he-il.facebook.com/groups/200491360554968).
For now the file saved in my cloud storage.
Do you need access to this file?

File: engines/sword25/gfx/graphicengine.cpp (hunk 1 above)
Comment (ShaharAriel) [reply engines/sword25/gfx/graphicengine.cpp]: Done

File: engines/sword25/gfx/graphicengine.h (hunk 3 above)
Comment (ShaharAriel) [reply engines/sword25/gfx/graphicengine.h]: Done

**Comment by ShaharAriel** [comment]
> Good stuff, but I have a few minor notes.

File: engines/sword25/gfx/text.cpp [hunk 4]
```diff
@@ -286,10 +292,14 @@ void Text::updateFormat() {
 			_height += bbox.height();
 		}
 	} else {
-		// No auto format, so all the text is copied to a single line.
-		_lines[0].text = _text;
-		_lines[0].bbox = Common::Rect(0, 0, _width, _height);
-	}
+        if (isRTL) {
+            _lines[0].text = Common::convertBiDiString(_text, Common::kWindows1255);
+        }else {
```
[the remark this answers is not in the document]
Comment (ShaharAriel) [reply engines/sword25/gfx/text.cpp]: Done, thank you all

File: engines/sword25/detection_tables.h (hunk 2 above)
Comment (sev-) [reply engines/sword25/detection_tables.h:111]: I'd like to have it later, so we put on our website.

**Comment by sev-** [comment]
Thank you

**Pull request merged**